base/setup/lib/fsutil.c

```diff
@@ -111,6 +111,35 @@ typedef struct _BTRFS_BOOTSECTOR
 } BTRFS_BOOTSECTOR, *PBTRFS_BOOTSECTOR;
 C_ASSERT(sizeof(BTRFS_BOOTSECTOR) == BTRFS_BOOTSECTOR_SIZE);
 
+typedef struct _NTFS_BOOTSECTOR
+{
+    UCHAR Jump[3];
+    UCHAR OEMID[8];
+    USHORT BytesPerSector;
+    UCHAR SectorsPerCluster;
+    UCHAR Unused0[7];
+    UCHAR MediaId;
+    UCHAR Unused1[2];
+    USHORT SectorsPerTrack;
+    USHORT Heads;
+    UCHAR Unused2[4];
```

Comment:
This field is actually "HiddenSectors", and required by the boot sector to find the begining sector of the NTFS partition.
Also, add a sanity/consistency check on that value before installing the boot sector would be nice.

For reference, https://www.ntfs.com/fat-boot-modif.htm

Reply:
I deliberately took this from a reliable source [reactos.git/drivers/filesystems/ntfs/ntfs.h](https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/filesystems/ntfs/ntfs.h;hb=2538583dbe13560676ac9bb3dc0911b8a070c1a1#l25)
Your link is for FAT32 and this is not the same thing.
From another very reliable source :) [How NTFS Works](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2003/cc781134(v=ws.10)) you can make sure that this field is "Not used or checked by NTFS".

Reply:
My apologies, it looks like I didn't paste the link I wanted too, so, here's the one I intended to post: https://www.ntfs.com/ntfs-partition-boot-sector.htm

Also, I'm surprised Microsoft says this field is "unused": I'm pretty sure Windows 2003 can't boot with this field not set or with a wrong value, and my bootsect relies on it too.